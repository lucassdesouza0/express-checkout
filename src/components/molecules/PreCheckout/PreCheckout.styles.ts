import styled from "@emotion/styled";

import mq from "utils/queries";

export const PreCheckoutContainer = styled("div")`
  display: flex;
  flex-flow: column wrap;
  flex-direction: column;
  overflow-y: auto;
  width: 100vw;
  box-shadow: 2px 2px 10px 10px #eee;
`;

export const Section = styled("div")`
  padding: 10px;

  button {
    width: 250px;
    align-self: center;
  }
`;
