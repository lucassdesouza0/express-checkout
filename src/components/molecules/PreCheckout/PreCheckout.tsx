import React from "react";
import { useTranslation } from "next-i18next";

import SectionHeader from "atoms/SectionHeader/SectionHeader";
import DetailsCard from "molecules/DetailsCard/DetailsCard";
import VariantsCard from "molecules/VariantsCard/VariantsCard";
import { ProductContext, ProductProps } from "pages/_app";

import { PreCheckoutContainer, Section } from "./PreCheckout.styles";
import { useEffect } from "react";

export interface PreCheckoutProps {
  product?: ProductProps;
}

const PreCheckout = ({ product }: PreCheckoutProps) => {
  const { t } = useTranslation("common");

  if (!product) return <></>;

  useEffect(() => {}, [product]);

  return (
    <PreCheckoutContainer data-testid="precheckout-section">
      <SectionHeader title={t("pre-checkout-header")} />
      <Section>
        <DetailsCard product={product} />
        <VariantsCard product={product} />
      </Section>
    </PreCheckoutContainer>
  );
};

export default PreCheckout;
