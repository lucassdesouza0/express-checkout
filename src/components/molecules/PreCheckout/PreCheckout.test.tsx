import React from "react";
import { screen, render, fireEvent } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import testRender from "../../../tests/render";

import mock from "../../../tests/__mocks__/product";

import PreCheckout from "./PreCheckout";

describe("Button", () => {
  test("it should mount", () => {
    testRender(<PreCheckout product={mock} />);

    const section = screen.getByTestId("precheckout-section");

    expect(section).toBeInTheDocument();
  });

  test("it should match snapshot", () => {
    const { asFragment } = render(<PreCheckout product={mock} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
