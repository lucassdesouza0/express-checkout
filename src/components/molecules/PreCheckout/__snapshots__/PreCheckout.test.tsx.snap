// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button it should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1jjeh5-PreCheckoutContainer erhl34w1"
    data-testid="precheckout-section"
  >
    <h2
      class="css-14jgswk-Header e2we4hg0"
    >
      pre-checkout-header
    </h2>
    <div
      class="css-1hnjv03-Section erhl34w0"
    >
      <div
        class="css-1oarku0-Card e1ss9q3t3"
        data-testid="details-card"
      >
        <img
          class="css-1bpjr25-SImage e65m3ty0"
          data-testid="image"
          height="100"
          src="https://images.prismic.io/barkyn/153c4252-8084-49b8-9931-3ec426cdd859_premium_box.png?auto=compress,format"
          width="100"
        />
        <div
          class="css-1lezaxk-Details e1ss9q3t2"
        >
          <p>
            Dry Dog Food - Chicken
          </p>
          <div
            class="css-sye7wj-Shipping e1ss9q3t0"
          />
        </div>
      </div>
      <div
        class="css-1ooee9y-Card e1a1q5s70"
        data-testid="variants-card"
      >
        <div
          class="css-1px0wn7-Section eui42cl2"
        >
          <div
            class="css-1l33m80-SelectionSection eui42cl1"
          >
            <label>
              packing-size:
            </label>
            <select
              class="css-1nbhqyd-Select eui42cl0"
              name="ages"
            >
              <option
                data-testid="select-option"
                disabled=""
                selected=""
                value="sizes"
              >
                sizes
              </option>
              <option
                data-testid="select-option-3kg"
                label="3kg"
                value="0"
              >
                3kg
              </option>
              <option
                data-testid="select-option-10kg"
                label="10kg"
                value="1"
              >
                10kg
              </option>
              <option
                data-testid="select-option-20kg"
                label="20kg"
                value="2"
              >
                20kg
              </option>
            </select>
          </div>
        </div>
        <button
          class="css-hhtkyx-Button enhitpr0"
          name="button"
          type="button"
        >
          <a
            href="/checkout"
          >
            kart
          </a>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
